sys/kern/signal.c

```diff
@@ -342,15 +347,27 @@ void sig_kill(proc_t *p, ksiginfo_t *ksi) {
     return;
   }
 
+  sigprop_t prop = sig_properties[sig];
+
+  /* Don't send TTY stop signals to members of an orphaned process group
+   * with a default handler.
+   * Theoretically, we should hold all_proc_mtx for reading pg_jobc,
+   * but the race here is harmless. */
+  if ((prop & SA_TTYSTOP) && (p->p_pgrp->pg_jobc == 0) && (handler == SIG_DFL))
```

Comment:
Maybe we can wrap this check and comment into simple function, because it repeats also in line 458.

Reply:
Done.